File: library/src/test/java/me/proxer/library/enums/GenderTest.java
package me.proxer.library.enums;

import me.proxer.library.ProxerTest;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author Ruben Gees
 */
class GenderTest extends ProxerTest {

    @Test
    void testDefault() throws IOException {
        final Gender gender = api.moshi().adapter(Gender.class).fromJson("\"f\"");

        assertThat(gender).isSameAs(Gender.FEMALE);
    }

    @Test
    void testFallback() throws IOException {
        final Gender gender = api.moshi().adapter(Gender.class).fromJson("\"xyz\"");

        assertThat(gender).isSameAs(Gender.UNKNOWN);
    }
}

File: library/src/test/kotlin/me/proxer/library/enums/GenderTest.kt
package me.proxer.library.enums

import com.serjltt.moshi.adapters.FallbackEnum
import com.squareup.moshi.Moshi
import org.amshove.kluent.shouldBe
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class GenderTest {

    private val adapter = Moshi.Builder()
        .add(FallbackEnum.ADAPTER_FACTORY)
        .build()
        .adapter(Gender::class.java)

    @Test
    fun testDefault() {
        adapter.fromJson("\"f\"") shouldBe Gender.FEMALE
    }

    @Test
    fun testFallback() {
        adapter.fromJson("\"xyz\"") shouldBe Gender.UNKNOWN
    }
}
